common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -395,37 +485,92 @@ public ICacheRecord load(@NonNull final String clientId,
         return result;
     }
 
+    @Override
+    public List<ICacheRecord> loadWithAggregatedAccountData(@NonNull final String clientId,
+                                                            @Nullable final String target,
+                                                            @NonNull final AccountRecord account) {
+        final List<ICacheRecord> result = new ArrayList<>();
+
+        final ICacheRecord primaryCacheRecord = load(clientId, target, account);
+
+        // Set this result as the 0th entry in the result...
+        result.add(primaryCacheRecord);
+
+        final List<ICacheRecord> corollaryCacheRecords = getAccountsWithAggregatedAccountData(
+                account.getEnvironment(),
+                clientId,
+                account.getHomeAccountId()
+        );
+
+        // corollaryCacheRecords will contain the original element that we've already added to
+        // our result so skip that element, but add the rest...
+        for (final ICacheRecord cacheRecord : corollaryCacheRecords) {
+            if (!account.equals(cacheRecord.getAccount())) {
+                result.add(cacheRecord);
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public List<IdTokenRecord> getIdTokensForAccount(@Nullable String clientId,
```

Comment:
I think Account in the internal sense vs Account in the public sense is likely to get confusing for us going foward.  We may want to align these two models of the world in our naming at some point to improve our chances of avoiding errors.

Reply:
Resolved in be3f9ffcba3e3f70e72b7b792c79f0e64039f0a7